src/workerd/api/http.c++

```diff
@@ -1893,19 +1905,15 @@ jsg::Promise<jsg::Ref<Response>> Fetcher::fetch(
   return fetchImpl(js, JSG_THIS, kj::mv(requestOrUrl), kj::mv(requestInit));
 }
 
-static jsg::Promise<void> throwOnError(
-    kj::StringPtr method, jsg::Promise<jsg::Ref<Response>> promise) {
-  return promise.then([method](jsg::Ref<Response> response) {
+static jsg::Promise<void> throwOnError(jsg::Lock& js,
+                                       kj::StringPtr method,
+                                       jsg::Promise<jsg::Ref<Response>> promise) {
+  return promise.then(js, [method](jsg::Lock&, jsg::Ref<Response> response) {
```

Comment:
co_await the promise?

Reply:
This is a `jsg::Promise`, co_await is not supported on these yet.